src/kinetics/Arrhenius.cpp

```diff
@@ -171,7 +175,7 @@ void TwoTempPlasmaRate::setRateParameters(
         ArrheniusBase::setRateParameters(rate, units, rate_units);
         auto& rate_map = rate.as<AnyMap>();
         // check Ea-gas value. Set to zero in the case of absent.
-        if (m_Ea_R == NAN) {
+        if (isnan(m_Ea_R)) {
```

Comment:
There's no circumstance where `m_Ea_R` is initialized to `NAN`, is there? If not, I don't think there's a reason to check for that condition.

Reply:
This is necessary to pass the test. I think it is due to m_Ea_R = NAN when the key is absent.

Reply:
I think we can table the discussion on this aspect. I hope to eliminate non-initialized `ReactionRate` objects altogether prior to releasing 2.6 (which #1183 will make possible), meaning that the `NAN` sentinels won't be necessary any longer. At the moment, the YAML serialization needs a valid rate object attached to preserve the type of the reaction. After #1183 (and follow-ups), specialization will only occur at the `ReactionRate` level.

Reply:
Ah, that's great to hear. I did realize, though that the case where this is getting set to `NAN` that needs to dealt with is the case "when the key is absent" as @BangShiuh said, which occurs in the call to the parent class's method, i.e. `ArrheniusBase::setRateParameters`, and so that does need to be dealt with.